internal/server/middleware.go

```diff
@@ -112,8 +113,40 @@ func authenticateRequest(c *gin.Context, route routeSettings, srv *Server) (acce
 		}
 	}
 
-	err = tx.Commit()
-	return authned, err
+	if err := tx.Commit(); err != nil {
+		return authned, err
+	}
+
+	if authned.User != nil && route.idpSync {
+		if route.authenticationOptional {
+			// this should be caught during development
+			return authned, fmt.Errorf("idp sync requires authentication")
+		}
+		tx2, err := srv.db.Begin(c.Request.Context(), nil)
+		if err != nil {
+			return authned, err
+		}
+		defer logError(tx2.Rollback, "failed to rollback identity provider sync transaction")
+		tx2 = tx2.WithOrgID(authned.Organization.ID)
+		// sync the identity info here to keep the UI session in sync with IDP session validity
+		if err := srv.syncIdentityInfo(context.Background(), tx2, authned.User, authned.AccessKey.ProviderID); err != nil {
+			deleteCookie(c.Writer, cookieAuthorizationName, c.Request.Host)
+			if errors.Is(err, ErrSyncFailed) {
+				logging.L.Debug().Err(err)
+			} else {
+				logging.L.Error().Err(err)
+			}
+			if err = tx2.Commit(); err != nil {
+				logging.L.Error().Err(err)
+			}
```

Comment:
It's a bit unconventional to commit in an error handling block, but I guess the reason this is needed is because we may have updated the ProviderUser, or delete access keys.

I can't think of any obvious problems this would cause. We'll have to keep this in mind as we make changes. It's possible we might want some more fine grained transaction handling in `syncIdentityInfo` and the functions it calls, but let's worry about that in the future.